craft_providers/util/temp_paths.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Helpers for temporary files."""
+
+import contextlib
+import pathlib
+import tempfile
+from typing import Iterator
+
+
+@contextlib.contextmanager
+def home_temporary_directory() -> Iterator[pathlib.Path]:
+    """Create temporary directory in home directory where Multipass has access."""
+    with tempfile.TemporaryDirectory(
+        suffix=".tmp-craft",
+        dir=pathlib.Path.home(),
+    ) as tmp_dir:
+        yield pathlib.Path(tmp_dir)
+
+
+@contextlib.contextmanager
+def home_temporary_file(name: str) -> Iterator[pathlib.Path]:
+    """Create temporary directory in home directory where Multipass has access.
+
+    :param name: Name of file to use (in temporary directory).
+    """
+    with home_temporary_directory() as tmp_dir:
+        yield tmp_dir / name
```

Comment:
We can also verify if the file already exists, but if we only return a path instead of a file we may  incur a TOCTOU race.

Reply:
Ah, well that shouldn't be a problem because the temp dir is guaranteed, and the caller is the sole user.

But based on your earlier comment, maybe it's worth just nuking this helper and stick with home_temporary_directory

Reply:
Removed for now.